tools/tsp-client/src/fs.ts

```diff
@@ -47,3 +48,48 @@ export async function tryReadTspLocation(rootDir: string): Promise<string | unde
   }
   return undefined;
 }
+
+export async function readTspLocation(rootDir: string): Promise<[string, string, string, string[]]> {
+  try {
+    const yamlPath = path.resolve(rootDir, "tsp-location.yaml");
+    const fileStat = await stat(yamlPath);
+    if (fileStat.isFile()) {
+      const fileContents = await readFile(yamlPath, "utf8");
+      const locationYaml = parseYaml(fileContents);
+      var { directory, commit, repo, additionalDirectories } = locationYaml;
+      if (!directory || !commit || !repo) {
+        throw new Error("Invalid tsp-location.yaml");
+      }
+      Logger.info(`Additional directories: ${additionalDirectories}`)
+      if (!additionalDirectories) {
+        additionalDirectories = [];
+      }
+      return [ directory, commit, repo, additionalDirectories ];
+    }
+  } catch (e) {
+    Logger.error(`Error reading tsp-location.yaml: ${e}`);
+  }
+  throw new Error("Could not find tsp-location.yaml");
+}
+
+
+export async function findEmitterPackage(emitterPath: string): Promise<string | undefined> {
```

Comment:
I think this method could be named a little better since what it appears to do is take the path to a repository `package.json` that *might* contain an emitter package dependency, but that's not very clear

Also it's a little strange that we resolve to a single exporter when theoretically the `emitter-package.json` could have multiple emitters. Should we support that?

Reply:
From what I read in the instructions for the powershell scripts, `emitter-package.json` is part of the required set up in the language repos. I would expect it to always be there, but can clarify the name of this function more. Maybe `findEmitterPackageJson`? Or `findEmitterPackageConfig`? Not sure about whether a language repo would support multiple emitters, at least not initially, we could work on adding this later if that's a possible use case

Reply:
Also, this was a workaround meanwhile I'm work on adding the custom language setting functionality that existed previously with the powershell scripts.

Reply:
I think what was confusing me is "findEmitterPackage" makes it sound like this is where the typespec emitter lives, rather than the repo config for which emitter package to use, so perhaps `getEmitterFromRepoConfig` or somesuch?